console/src/utils/migration/utils.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  generateTableDef,
+  findTable,
+  getTableCustomColumnNames,
+  getTableCustomRootFields,
+} from '../../components/Common/utils/pgUtils';
+import { isColumnUnique } from '../../components/Services/Data/TableModify/ModifyActions';
+import {
+  getRunSqlQuery,
+  getSetCustomRootFieldsQuery,
+} from '../../components/Common/utils/v1QueryUtils';
+import {
+  isColTypeString,
+  isPostgresFunction,
+} from '../../components/Services/Data/utils';
+import gqlPattern from '../../components/Services/Data/Common/GraphQLValidation';
+import { sqlEscapeText } from '../../components/Common/utils/sqlUtils';
+import Migration from './Migration';
+import { RunSqlType } from '../../';
+
+// Types
+export interface NewColumnType {
+  tableName: string;
+  type: string;
+  isNullable: boolean;
+  isUnique: boolean;
+  default?: string;
+  comment?: string;
+  name: string;
+  schemaName?: string;
+  customFieldName?: string;
+}
+
+export interface SchemaType {
+  table_name: string;
+  table_type: string;
+  table_schema: string;
+}
+
+export interface OldColumnType {
+  udt_name: string;
+  data_type: string;
+  column_default: string | null;
+  comment: string | null;
+  is_nullable?: string;
+}
+
+// utils
+
+const parseNewCol = (newColumn: NewColumnType) => ({
+  tableName: newColumn.tableName,
+  colType: newColumn.type,
+  nullable: newColumn.isNullable,
+  unique: newColumn.isUnique,
+  colDefault: (newColumn.default || '').trim(),
+  comment: (newColumn.comment || '').trim(),
+  newName: newColumn.name.trim(),
+  currentSchema: newColumn.schemaName,
+  customFieldName: (newColumn.customFieldName || '').trim(),
+});
+
+const parseOldColumns = (oldColumn: OldColumnType) => ({
+  originalColType: oldColumn.udt_name, // "value"
+  originalData_type: oldColumn.data_type, // "value"
+  originalColDefault: oldColumn.column_default || '', // null or "value"
+  originalColComment: oldColumn.comment || '', // null or "value"
+  originalColNullable: oldColumn.is_nullable, // "YES" or "NO"
+});
+
+// utility to compare old & new columns and generate up &down migrations
+export const getColumnUpdateMigration = (
+  oldColumn: OldColumnType,
+  newColumn: NewColumnType,
+  allSchemas: [],
```

Comment:
We should never use array type this way, because it means literally an empty array. Check out this: https://www.typescriptlang.org/play/#code/DYUwLgBAHgXBDaBdA3AKFVCBeBBGANAEz4DMiQA

Schedules Triggers PR introduces proper typings for it, for now, you can use `unknown[]` or `object[]`.